Upstream: Yes, extracted from larger commit
Reason: Fixes runtime issues

diff --git a/main.c b/main.c
index d961ef0..ab9e4e0 100644
--- a/main.c
+++ b/main.c
@@ -5228,6 +5228,9 @@ main (int argc, char *argv[])
       if (path == NULL)
         goto err_out1;
       mkdir (path, 0700);
+      path = realloc(path, strlen(path)+strlen("/work")+1);
+      if (!path)
+        error (EXIT_FAILURE, errno, "allocating workdir path");
       strcat (path, "/work");
       mkdir (path, 0700);
       free (lo.workdir);
diff --git a/utils.c b/utils.c
index 43073f1..981fe86 100644
--- a/utils.c
+++ b/utils.c
@@ -98,7 +98,12 @@ safe_openat (int dirfd, const char *pathname, int flags, mode_t mode)
 int
 file_exists_at (int dirfd, const char *pathname)
 {
-  return faccessat (dirfd, pathname, F_OK, AT_SYMLINK_NOFOLLOW|AT_EACCESS);
+  int ret = faccessat (dirfd, pathname, F_OK, AT_SYMLINK_NOFOLLOW|AT_EACCESS);
+  if (ret < 0 && errno == EINVAL) {
+    struct stat buf;
+    return fstatat (dirfd, pathname, &buf, AT_SYMLINK_NOFOLLOW);
+  }
+  return ret;
 }
 
 #ifdef HAVE_STATX
